cognite/client/data_classes/postgres_gateway/tables.py

```diff
@@ -0,0 +1,332 @@
+        self.options = options
+        self.columns: ColumnList = ColumnList(columns)
+
+    @classmethod
+    def _load_table(cls, data: dict[str, Any]) -> Self:
+        return cls(
+            tablename=data["tablename"],
+            options=RawTableOptions._load(data["options"]),
+            columns=ColumnList._load_columns(data["columns"]),
+        )
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        output = super().dump(camel_case=camel_case)
+        output["options"] = self.options.dump(camel_case=camel_case)
+        output["columns"] = self.columns._dump_columns()
+        return output
+
+
+class ViewTableWrite(TableWrite):
+    """Foreign tables.
+
+    This is the read/response format of the custom table.
+
+    Args:
+        tablename (str): Name of the foreign table.
+        options (ViewId): Table options
+    """
+
+    _type = "view"
+
+    def __init__(self, tablename: str, options: ViewId) -> None:
+        super().__init__(tablename=tablename)
+        self.options = options
+
+    @classmethod
+    def _load_table(cls, data: dict[str, Any]) -> Self:
+        return cls(
+            tablename=data["tablename"],
+            options=ViewId.load(data["options"]),
+        )
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        output = super().dump(camel_case=camel_case)
+        output["options"] = self.options.dump(camel_case=camel_case, include_type=False)
+        return output
+
+
+class Table(_TableCore, ABC):
+    """Foreign tables.
+
+    This is the read/response format of the custom table.
+
+    Args:
+        tablename (str): Name of the foreign table.
+        created_time (int | None): Time when the table was created
+
+    """
+
+    def __init__(self, tablename: str, created_time: int | None = None) -> None:
+        super().__init__(tablename=tablename)
+        self.created_time = created_time
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        type_ = resource.get("type")
+        if type_ is None and hasattr(cls, "_type"):
+            type_ = cls._type
+        elif type_ is None:
+            raise KeyError("type")
+        table_cls = _TABLE_CLASS_BY_TYPE.get(type_)
+        if table_cls is None:
+            return UnknownCogniteObject(resource)  # type: ignore[return-value]
+        return cast(Self, table_cls._load_table(resource))
+
+    @classmethod
+    @abstractmethod
+    def _load_table(cls, data: dict[str, Any]) -> Self:
+        raise NotImplementedError()
+
+
+class RawTable(Table):
+    """Foreign tables.
+
+    This is the read/response format of the raw table.
+
+    Args:
+        tablename (str): Name of the foreign table.
+        options (RawTableOptions): Table options
+        columns (ColumnList): Foreign table columns.
+        created_time (int | None): Time when the table was created.
+
+    """
+
+    _type = "raw_rows"
```

Comment:
This is also used by `RawTableWrite` - intended?

Reply:
I don't think I follow here. RawTableWrite also has type set

Reply:
Just that they used the same value of `_type`, but I guess it's okay as they are subclasses of different classes